src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitThis = sourceString.split(';');
+  const result = {};
+
+  for (const oneString of splitThis) {
+    let word = '';
+    let wordTwo = '';
+    let fazeTwo = false;
+
+    for (const symbol of oneString) {
+      if (symbol.toLowerCase() !== symbol.toUpperCase() || symbol === '-') {
+        if (fazeTwo !== true) {
+          word += symbol;
+        }
+      }
+
+      if (fazeTwo === true) {
+        wordTwo += symbol;
+      }
+
+      if (symbol === ':') {
+        fazeTwo = true;
+      }
+    }
+
+    if (word.length !== 0) {
+      result[word] = wordTwo.trim();
```

Comment:
The check for `word.length !== 0` is to ensure that empty strings are not added as keys to the result object. However, this can be handled more elegantly by filtering out empty strings earlier in the process, for example, by using `filter(Boolean)` after the initial `split` on semicolons.